addons/website/tests/test_website_visitor.py

```diff
@@ -101,6 +101,10 @@ def setUp(self):
                 'partner_id': self.partner_portal.id,
                 'groups_id': [(6, 0, [self.env.ref('base.group_portal').id])],
             })
+        self.partner_demo1 = self.env['res.partner'].create({
```

Comment:
I find `partner_demo1` confusing, as it is not at all related to Marc Demo, right? Especially since you merge it with Admin...
How about something like `partner_admin_duplicate`, `partner_admin_to_merge`, or `partner_admin_other`?
As it is an IMP, with breaking behaviors, you can be more thorough in your refactoring of existing test, no need to keep a "demo" reference if it doesn't make sense anymore? IMO it's worth it the a few additional lines in diff.